sympy/solvers/simplex.py

```diff
@@ -0,0 +1,821 @@
+            else:
+                raise TypeError('only Ge, Le, or Eq allowed, not %s' % i)
+        if i == True:
+            continue  # ignore
+        if i == False:
+            return  # no solution
+        if isinstance(i, Eq):
+            # could also collect k equalities and solve for
+            # k variables and eliminate them
+            L = i.lhs - i.rhs
+            np.extend([L, -L])
+            continue
+
+        i = i.canonical  # +/-oo, if present will be on the rhs
+        if isinstance(i, (Le, Ge, Lt, Gt)) and i.rhs.is_infinite:
+            if isinstance(i, (Lt, Le)) and i.rhs is S.NegativeInfinity:
+                return False
+            if isinstance(i, (Gt, Ge)) and i.rhs is S.Infinity:
+                return False
+            unbound.update(i.lhs.free_symbols) # x < oo, x > -oo
+        elif isinstance(i, (Le, Ge)):
+            npi = i.lts - i.gts
+            x = npi.free_symbols
+            if len(x) == 1:
+                x = x.pop()
+                if x in unbound:
+                    continue  # will handle later
+                ivl = Le(npi, 0, evaluate=False).as_set()
+                if x not in univariate:
+                    univariate[x] = ivl
+                else:
+                    univariate[x] &= ivl
+            else:
+                np.append(npi)
+        else:
+            raise TypeError('only Ge, Le, or Eq is allowed, not %s' % i)
+
+    # introduce auxilliary variables as needed for univariate
+    # inequalities
+    for x in univariate:
+        i = univariate[x]
+        if not i:
+            return None  # no solution possible
+        a, b = i.inf, i.sup
+        if a.is_infinite and b.is_infinite:
+            unbound.append(x)
+        elif a.is_infinite:
+            u = next(ui)
+            r[x] = b - u
+            aux.append(u)
+        elif b.is_infinite:
+            if a:
+                u = next(ui)
+                r[x] = a + u
+                aux.append(u)
+            else:
+                # standard nonnegative relationship
+                pass
+        else:
+            u = next(ui)
+            aux.append(u)
+            # shift so u = x - a => x = u + a
+            r[x] = u + a
+            # add constraint for u <= b - a
+            # since when u = b-a then x = u + a = b - a + a = b:
+            # the upper limit for x
+            np.append(u - (b - a))
+
+    # make change of variables for unbound variables
+    for x in unbound:
+        u = next(ui)
+        r[x] = u - x  # reusing x
+        aux.append(u)
+
+    return np, r, aux
+
+
+def _lp_matrices(objective, constraints):
+    """return A, B, C, D, r, x+X, X for maximizing
+    objective = Cx - D with constraints Ax <= B, introducing
+    introducing auxilliary variables, X, as necessary to make
+    replacements of symbols as given in r, {xi: expression with Xj},
+    so all variables in x+X will take on nonnegative values.
+
+    If symbols not in objective will be set to 0.
```

Comment:
What if:

Minimize x subject to y = 2,  x=y

If we get rid of y, then the constraint x=y becomes meaningless.

Reply:
That would require resolving equalities, favoring those that don't appear in the objective. I preferred that, anyway, and can look at making that change.

Reply:
The example I gave involved equalities, but constraints involving variables not in the objective function can also be meaningful in situations without equalities: 

Minimize x subject to y >= 10, x >= y

Here the problem becomes unbounded when you remove y even though originally this wasn't the case.

Reply:
This still doesn't work: `lpmin(x,[y >= 10, x >= y])` -> Infeasible